pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,101 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			var linkableCluster config.Cluster
+			for _, cluster := range ctx.Clusters() {
+				if args[0] == cluster.ID() ||
+					args[0] == cluster.URL() ||
```

Comment:
Currently `dcos cluster link` only supports passing an URL, it requires the cluster to already be configured in the CLI though. Originally the idea was that, in order to link cluster A to cluster B, assuming A is already configured and attached, instead of having to: 
1) setup B
2) attach back to A
3) link A to B

One could just do `dcos cluster link B` using its URL, and this would trigger a setup "on-the-fly" before it eventually links A to B at the end of the flow. We didn't implement it as this required a lot of refactoring on the current CLI (to allow a setup on the fly) and thought we'd tackle this during this refactoring.

I hope we can all agree that the current implementation has a very bad UX, configured clusters should only be referred to using their name or ID, not URL. Here I think `dcos cluster link <url>` should trigger a cluster setup, and `dcos cluster link <id_or_name>` should look-up an already configured cluster.

@vinodkone : could you please confirm this one? I remember we already discussed this.

Reply:
I've added the new workflow.